### PR TITLE
Added feature to generate png on save

### PR DESCRIPTION
**Pull request by onoufriosm**

### CODE REVIEW

File: src/withEditorProps/index.js [hunk 1]
```diff
@@ -23,11 +25,57 @@ import { allTypes } from "../utils/annotationTypes";
...
+ * Adds a div in the dom and
+ * renders ComponentToPrint inside
+ *
+ * @return {object} - div element
+ */
+const generateComponentToPrint = () => {
+  const componentToPrint = document
+    .querySelector("div.veVectorInteractionWrapper")
+    .cloneNode(true);
+  // const componentToPrint = <ComponentToPrint editorName="" circular />; // TODO: Pass correct editorName
```
Comment (onoufriosm) [inline src/withEditorProps/index.js]: Need to uncomment this line. Atm I'm just cloning the active view (circular/linear) in the editor.

File: src/withEditorProps/index.js [hunk 2]
```diff
@@ -23,11 +25,57 @@ import { allTypes } from "../utils/annotationTypes";
...
+
+  const componentToPrintElement = document.createElement("div");
+  componentToPrintElement.setAttribute("id", "print-preview");
+  componentToPrintElement.style.width = "min-content";
+
+  // Making overflow hidden so a scrollbar doesn't appear
+  document.body.style.overflow = "hidden";
+
+  document.body.appendChild(componentToPrintElement);
+  // render(componentToPrint, componentToPrintElement);
```
Comment (onoufriosm) [inline src/withEditorProps/index.js]: Need to uncomment this line to render ComponentToPrint

File: src/withEditorProps/index.js [hunk 3]
```diff
@@ -42,19 +90,51 @@ export const handleSave = props => (opts = {}) => {
...
+      );
+
+    if (promiseOrVal && promiseOrVal.then) {
+      return promiseOrVal.then(updateLastSavedIdToCurrent);
+    }
+  };
+
+  // We don't want to make save slower for users that don't want
+  // a png as output
+  // if (!generatePng) {
```
Comment (onoufriosm) [inline src/withEditorProps/index.js]: This will be the option to pass to the Editor component to generate a png onsave.

**Comment by tnrich** [comment]
Hey @onoufriosm looking good! I added some code in there to pop up a customized PrintDialog component and then return it to the little generatePng hook you're working on. You can see that work here: https://github.com/TeselaGen/openVectorEditor/tree/pr/onoufriosm/598

I'll let you fix up the png generating functionality which I disabled. 

Let me know if you have anymore questions!

**Pull request closed without merging**